sdk/identity/identity/src/msal/nodeFlows/nodeCommon.ts

```diff
@@ -144,7 +181,7 @@ export abstract class MsalNode extends MsalBaseUtilities implements MsalFlow {
     if (this.account) {
       return this.account;
     }
-    const cache = this.publicApp?.getTokenCache();
+    const cache = this.confidentialApp?.getTokenCache() ?? this.publicApp?.getTokenCache();
```

Comment:
might it be easier to make a helper that returns either `this.confidentialApp` or `this.publicApp` as is needed instead of having to in-line this in a few places? e.g. `this.getApp()` or somesuch

Reply:
Having both `confidentialApp` and `publicApp` as properties of the clients is a code smell in and of itself, I think. I didn't want to try too hard to untangle that in this PR, and by adding a helper I think I might be compelled to go back through this class as well as its implementors to change them to use the getter. I would rather take this as an issue if that's alright.

Reply:
I'm good with capturing it in an issue! :)